src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/ClosureModelTranslator.groovy

```diff
@@ -101,28 +105,21 @@ public class ClosureModelTranslator implements MethodMissingWrapper, Serializabl
         } else {
             def resultValue
             def actualFieldName = Utils.actualFieldName(actualClass, methodName)
+            def actualType = Utils.actualFieldType(actualClass, methodName)
 
             // We care about the field name actually being a thing.
             if (actualFieldName != null) {
-                def actualType = Utils.actualFieldType(actualClass, methodName)
-
-                // Handle StepBlockWithOtherArgs *first*, since we won't recurse at all on them.
-                // If the field is an implementation of StepBlockWithOtherArgs, we need to just call its constructor with the args.
-                // Note that only Stage is a StepBlockWithOtherArgs currently, but that may be reused later.
-                if (Utils.assignableFromWrapper(StepBlockWithOtherArgs.class, actualType)) {
+                // Due to Stage taking an argument, not just a closure, we need to handle it differently.
+                if (Utils.assignableFromWrapper(Stage.class, actualType)) {
                     Object[] origArgs = args
-                    List<Object> blockParams = []
-                    for (int i = 0; i < origArgs.size(); i++) {
-                        def thisArg = origArgs[i]
-                        // If the argument is a Closure, create a StepsBlock of it.
-                        if (Utils.instanceOfWrapper(Closure.class, thisArg)) {
-                            blockParams.add(createStepsBlock(thisArg))
-                        } else {
-                            // Otherwise, just add the parameter.
-                            blockParams.add(thisArg)
-                        }
+                    String n = origArgs[0]
+                    def ctm = new ClosureModelTranslator(actualType, script)
+
+                    if (Utils.instanceOfWrapper(Closure.class, origArgs[1])) {
```

Comment:
Risk of `ArrayIndexOutOfBounds` if someone has a brain fart and writes something like.
```groovy
stage "build"
steps {
...
}
```